Sources/StreamVideo/Utils/AudioSession/RTCAudioStore/Components/AVAudioSessionObserver.swift

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Combine
+import Foundation
+
+extension AVAudioSession {
+    struct Snapshot: Equatable, CustomStringConvertible {
+        var category: AVAudioSession.Category
+        var mode: AVAudioSession.Mode
+        var categoryOptions: AVAudioSession.CategoryOptions
+        var routeSharingPolicy: AVAudioSession.RouteSharingPolicy
+        var availableModes: [AVAudioSession.Mode]
+        var preferredInput: RTCAudioStore.StoreState.AudioRoute.Port?
+        var renderingMode: String
+        var prefersEchoCancelledInput: Bool
+        var isEchoCancelledInputEnabled: Bool
+        var isEchoCancelledInputAvailable: Bool
+        var outputMuted: Bool
+        var maximumOutputNumberOfChannels: Int
+        var outputNumberOfChannels: Int
+        var preferredOutputNumberOfChannels: Int
+
+        var description: String {
+            var result = "{"
+            result += "category:\(category)"
+            result += ", mode:\(mode)"
+            result += ", categoryOptions:\(categoryOptions)"
+            result += ", routeSharingPolicy:\(routeSharingPolicy)"
+            result += ", availableModes:\(availableModes)"
+            result += ", preferredInput:\(preferredInput)"
+            result += ", renderingMode:\(renderingMode)"
+            result += ", prefersEchoCancelledInput:\(prefersEchoCancelledInput)"
+            result += ", isEchoCancelledInputEnabled:\(isEchoCancelledInputEnabled)"
+            result += ", isEchoCancelledInputAvailable:\(isEchoCancelledInputAvailable)"
+            result += ", outputMuted:\(outputMuted)"
+            result += ", maximumOutputNumberOfChannels:\(maximumOutputNumberOfChannels)"
+            result += ", outputNumberOfChannels:\(outputNumberOfChannels)"
+            result += ", preferredOutputNumberOfChannels:\(preferredOutputNumberOfChannels)"
+            result += " }"
+            return result
+        }
+
+        init(_ source: AVAudioSession = .sharedInstance()) {
+            self.category = source.category
+            self.mode = source.mode
+            self.categoryOptions = source.categoryOptions
+            self.routeSharingPolicy = source.routeSharingPolicy
+            self.availableModes = source.availableModes
+            self.preferredInput = source.preferredInput.map { .init($0) } ?? nil
+            if #available(iOS 17.2, *) { self.renderingMode = "\(source.renderingMode)" }
```

Comment:
we should test this stuff on older iOS versions

Reply:
Yes, would prefer to delete them if they are not used. They might confuse us 2 years from now.